include/ignition/gazebo/detail/View.hh

```diff
@@ -143,192 +151,73 @@ class View : public BaseView
   /// \sa invalidData
   private: std::unordered_map<Entity, std::unordered_set<ComponentTypeId>>
              missingCompTracker;
-};
-
-//////////////////////////////////////////////////
-template<typename ...ComponentTypeTs>
-View<ComponentTypeTs...>::View()
-{
-  this->componentTypes = {ComponentTypeTs::typeId...};
-}
 
-//////////////////////////////////////////////////
-template<typename ...ComponentTypeTs>
-bool View<ComponentTypeTs...>::HasCachedComponentData(
-    const Entity _entity) const
-{
-  auto cachedComps =
-    this->validData.find(_entity) != this->validData.end() ||
-    this->invalidData.find(_entity) != this->invalidData.end();
-  auto cachedConstComps =
-    this->validConstData.find(_entity) != this->validConstData.end() ||
-    this->invalidConstData.find(_entity) != this->invalidConstData.end();
-
-  if (cachedComps && !cachedConstComps)
+  template <typename... Args, std::size_t... Is, typename DataType>
+  static auto CreateTupleImpl(std::index_sequence<Is...>,
+                              const DataType &_args)
   {
-    ignwarn << "Non-const component data is cached for entity " << _entity
-      << ", but const component data is not cached." << std::endl;
+    return std::make_tuple(static_cast<Args>(_args[Is])...);
   }
-  else if (cachedConstComps && !cachedComps)
+
+  template <typename... Args, typename DataType>
+  static auto CreateTuple(const DataType &_args)
   {
-    ignwarn << "Const component data is cached for entity " << _entity
-      << ", but non-const component data is not cached." << std::endl;
+    return CreateTupleImpl<Args...>(std::index_sequence_for<Args...>{}, _args);
   }
-
-  return cachedComps && cachedConstComps;
-}
+};
 
 //////////////////////////////////////////////////
-template<typename ...ComponentTypeTs>
-bool View<ComponentTypeTs...>::RemoveEntity(const Entity _entity)
+/// Helper function to create a View out of template arguments
+template <typename... ComponentTypeTs>
+View MakeView()
```

Comment:
I've removed it. It was just for convenience, so that the changes in BaseView_TEST.cc can be minimized. https://github.com/ignitionrobotics/ign-gazebo/pull/1317/commits/16052878db2f1516a32ef6e4e5882d9eacba2005. But it makes sense to use the constructor directly.